### PR TITLE
Qa 2136 condensed cohort creation

### PR DESCRIPTION
**Pull request by bpierce01**
## Description
- Condensed test step for creating and saving a cohort with specified filters
- Test step for for creating and saving a cohort with randomized filters

### CODE REVIEW

**Comment by bpierce01** [comment]
Let me know if you have any questions about the logic/loops for the randomized steps. I tried to leave comments explaining all the steps.

File: holmes-py/step_impl/apps/gdc_data_portal_v2/pages/cohort_builder_page.py [hunk 1]
```diff
@@ -120,3 +134,73 @@ def is_query_expression_area_text_present(self, text):
...
+        able to select by position. Conditions: the facet card uses checkboxes, and there
+        is available data present to select.
+
+        :param facet_to_select_by_position: The position of the facet to analyze.
+        :return: True or False is the facet card able to select by position.
+        """
+        filter_list_have_checkbox_locator = CohortBuilderPageLocators.FILTER_LIST_CHECKBOX(facet_to_select_by_position)
+        if self.is_visible(filter_list_have_checkbox_locator):
+            no_data_available_locator = CohortBuilderPageLocators.FILTER_LIST_NO_DATA_AVAILABLE(facet_to_select_by_position)
+            if self.is_visible(no_data_available_locator):
```
Comment (uc-pperumala) [inline holmes-py/step_impl/apps/gdc_data_portal_v2/pages/cohort_builder_page.py]: Use one-liner: `return False if self.is_visible(no_data_available_locator) else True`

**Review by uc-pperumala — CHANGES_REQUESTED** [review]
```
# Cohort Bar - Save, Switch, Delete Cohort
  ## Navigate to Cohort Builder  ✔ ✔
  ## Save a cohort for the first time    ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔
  ## Save a cohort for the second time   ✔ ✔ ✔ ✔ ✘
        Failed Step: Perform action and validate modal text
        Specification: specs/gdc_data_portal_v2/cohort_bar/general_cohort_functions.spec:41
        Error Message: The text 'Cohort has been saved' is NOT present
        Stacktrace: 
        Traceback (most recent call last):
          File "/Users/pramodperumala/.local/lib/python3.8/site-packages/getgauge/executor.py", line 33, in execute_method
            step.impl(*params)
          File "/Users/pramodperumala/Software/gdc-frontend-framework/holmes-py/step_impl/apps/gdc_data_portal_v2/step_defs/generic_steps.py", line 513, in click_named_button_in_modal_and_wait_for_temp_message_text
            assert is_cohort_message_present, f"The text '{v[1]}' is NOT present"
        AssertionError: The text 'Cohort has been saved' is NOT present
        
  ## Switch cohorts      ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔
  ## Delete a saved cohort       ✔ ✔ ✔ ✔
  ## Delete an unsaved cohort    ✔ ✔ ✔ ✔ ✔
  ## Random filters cohort creation      ✔
  ## Shortened cohort saving     ✔

Successfully generated html-report to => /Users/pramodperumala/Software/gdc-frontend-framework/holmes-py/reports/html-report/index.html

Specifications: 1 executed      0 passed        1 failed        0 skipped
Scenarios:      8 executed      7 passed        1 failed        0 skipped

Total time taken: 3m43.655s
Updates are available. Run `gauge update -c` for more info.
(venv) pramodperumala@Pramods-MacBook-Pro holmes-py % gauge run specs/gdc_data_portal_v2/cohort_bar/general_cohort_functions.spec

Python: 3.8.9
E0214 20:03:59.155054000 123145513623552 fork_posix.cc:76]             Other threads are currently calling into gRPC, skipping fork() handlers
E0214 20:04:01.595433000 123145513623552 fork_posix.cc:76]             Other threads are currently calling into gRPC, skipping fork() handlers
# Cohort Bar - Save, Switch, Delete Cohort
  ## Navigate to Cohort Builder  ✔ ✔
  ## Save a cohort for the first time    ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔
  ## Save a cohort for the second time   ✔ ✔ ✔ ✔ ✘
        Failed Step: Perform action and validate modal text
        Specification: specs/gdc_data_portal_v2/cohort_bar/general_cohort_functions.spec:41
        Error Message: The text 'Cohort has been saved' is NOT present
        Stacktrace: 
        Traceback (most recent call last):
          File "/Users/pramodperumala/.local/lib/python3.8/site-packages/getgauge/executor.py", line 33, in execute_method
            step.impl(*params)
          File "/Users/pramodperumala/Software/gdc-frontend-framework/holmes-py/step_impl/apps/gdc_data_portal_v2/step_defs/generic_steps.py", line 513, in click_named_button_in_modal_and_wait_for_temp_message_text
            assert is_cohort_message_present, f"The text '{v[1]}' is NOT present"
        AssertionError: The text 'Cohort has been saved' is NOT present
        
  ## Switch cohorts      ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔
  ## Delete a saved cohort       ✔ ✔ ✔ ✔
  ## Delete an unsaved cohort    ✔ ✔ ✔ ✔ ✔
  ## Random filters cohort creation      ✔
  ## Shortened cohort saving     ✔

Successfully generated html-report to => /Users/pramodperumala/Software/gdc-frontend-framework/holmes-py/reports/html-report/index.html

Specifications: 1 executed      0 passed        1 failed        0 skipped
Scenarios:      8 executed      7 passed        1 failed        0 skipped

Total time taken: 3m29.172s
```

Seeing some failures (1 step failing for multiple scenarios)

File: holmes-py/step_impl/apps/gdc_data_portal_v2/pages/cohort_builder_page.py (hunk 1 above)
Comment (bpierce01) [reply holmes-py/step_impl/apps/gdc_data_portal_v2/pages/cohort_builder_page.py]: Addressed!

**Comment by bpierce01** [comment]
> ```
> # Cohort Bar - Save, Switch, Delete Cohort
>   ## Navigate to Cohort Builder  ✔ ✔
>   ## Save a cohort for the first time    ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔
>   ## Save a cohort for the second time   ✔ ✔ ✔ ✔ ✘
>         Failed Step: Perform action and validate modal text
>         Specification: specs/gdc_data_portal_v2/cohort_bar/general_cohort_functions.spec:41
>         Error Message: The text 'Cohort has been saved' is NOT present
>         Stacktrace: 
>         Traceback (most recent call last):
>           File "/Users/pramodperumala/.local/lib/python3.8/site-packages/getgauge/executor.py", line 33, in execute_method
>             step.impl(*params)
>           File "/Users/pramodperumala/Software/gdc-frontend-framework/holmes-py/step_impl/apps/gdc_data_portal_v2/step_defs/generic_steps.py", line 513, in click_named_button_in_modal_and_wait_for_temp_message_text
>             assert is_cohort_message_present, f"The text '{v[1]}' is NOT present"
>         AssertionError: The text 'Cohort has been saved' is NOT present
>         
>   ## Switch cohorts      ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔
>   ## Delete a saved cohort       ✔ ✔ ✔ ✔
>   ## Delete an unsaved cohort    ✔ ✔ ✔ ✔ ✔
>   ## Random filters cohort creation      ✔
>   ## Shortened cohort saving     ✔
> 
> Successfully generated html-report to => /Users/pramodperumala/Software/gdc-frontend-framework/holmes-py/reports/html-report/index.html
> 
> Specifications: 1 executed      0 passed        1 failed        0 skipped
> Scenarios:      8 executed      7 passed        1 failed        0 skipped
> 
> Total time taken: 3m43.655s
> Updates are available. Run `gauge update -c` for more info.
> (venv) pramodperumala@Pramods-MacBook-Pro holmes-py % gauge run specs/gdc_data_portal_v2/cohort_bar/general_cohort_functions.spec
> 
> Python: 3.8.9
> E0214 20:03:59.155054000 123145513623552 fork_posix.cc:76]             Other threads are currently calling into gRPC, skipping fork() handlers
> E0214 20:04:01.595433000 123145513623552 fork_posix.cc:76]             Other threads are currently calling into gRPC, skipping fork() handlers
> # Cohort Bar - Save, Switch, Delete Cohort
>   ## Navigate to Cohort Builder  ✔ ✔
>   ## Save a cohort for the first time    ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔
>   ## Save a cohort for the second time   ✔ ✔ ✔ ✔ ✘
>         Failed Step: Perform action and validate modal text
>         Specification: specs/gdc_data_portal_v2/cohort_bar/general_cohort_functions.spec:41
>         Error Message: The text 'Cohort has been saved' is NOT present
>         Stacktrace: 
>         Traceback (most recent call last):
>           File "/Users/pramodperumala/.local/lib/python3.8/site-packages/getgauge/executor.py", line 33, in execute_method
>             step.impl(*params)
>           File "/Users/pramodperumala/Software/gdc-frontend-framework/holmes-py/step_impl/apps/gdc_data_portal_v2/step_defs/generic_steps.py", line 513, in click_named_button_in_modal_and_wait_for_temp_message_text
>             assert is_cohort_message_present, f"The text '{v[1]}' is NOT present"
>         AssertionError: The text 'Cohort has been saved' is NOT present
>         
>   ## Switch cohorts      ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔ ✔
>   ## Delete a saved cohort       ✔ ✔ ✔ ✔
>   ## Delete an unsaved cohort    ✔ ✔ ✔ ✔ ✔
>   ## Random filters cohort creation      ✔
>   ## Shortened cohort saving     ✔
> 
> Successfully generated html-report to => /Users/pramodperumala/Software/gdc-frontend-framework/holmes-py/reports/html-report/index.html
> 
> Specifications: 1 executed      0 passed        1 failed        0 skipped
> Scenarios:      8 executed      7 passed        1 failed        0 skipped
> 
> Total time taken: 3m29.172s
> ```
> 
> Seeing some failures (1 step failing for multiple scenarios)

Yes that is a known and expected failure. There is a front-end ticket for a fix.

**Review by uc-pperumala — APPROVED** [review]
Approved.

**Review by wteouchicago — APPROVED** [review]

**Pull request merged**